### ISSUE TITLE
Not compatible with gjs 1.75.2

### ISSUE THREAD

**Issue by jbicha** [micheleg/dash-to-dock#1919]
- gjs 1.75.2 (part of GNOME 44 Beta)
- Ubuntu 23.04
- gnome-shell-extension-ubuntu-dock 75ubuntu1

When I try to activate the extension with gjs 1.75.2 (currently in Debian Experimental), I get this error:

`Func get_id returns a transfer-none string. This is not supported ( https://gitlab.gnome.org/GNOME/gjs/-/issues/519 )`

The extension works normally with the stable gjs 1.74.1 or 1.74.2.

**Comment by vanvugt** [comment]
```
(gnome-shell:125881): Gjs-WARNING **: 14:41:10.066: JS ERROR: Extension ubuntu-dock@ubuntu.com: Error: VFunc get_id returns a transfer-none string. This is not supported (https://gitlab.gnome.org/GNOME/gjs/-/issues/519)
registerGObjectType/<@resource:///org/gnome/gjs/modules/core/overrides/GObject.js:566:53
registerGObjectType@resource:///org/gnome/gjs/modules/core/overrides/GObject.js:558:10
_init/GObject.Object._classInit@resource:///org/gnome/gjs/modules/core/overrides/GObject.js:515:33
registerClass@resource:///org/gnome/gjs/modules/core/overrides/GObject.js:74:23
@/usr/share/gnome-shell/extensions/ubuntu-dock@ubuntu.com/locations.js:80:31
@/usr/share/gnome-shell/extensions/ubuntu-dock@ubuntu.com/appIcons.js:36:19
@/usr/share/gnome-shell/extensions/ubuntu-dock@ubuntu.com/dash.js:24:18
@/usr/share/gnome-shell/extensions/ubuntu-dock@ubuntu.com/docking.js:35:18
@/usr/share/gnome-shell/extensions/ubuntu-dock@ubuntu.com/extension.js:6:17
_callExtensionInit@resource:///org/gnome/shell/ui/extensionSystem.js:478:13
async*loadExtension@resource:///org/gnome/shell/ui/extensionSystem.js:394:33
_loadExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:677:24
async*_enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:683:48
_sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:718:20
async*init@resource:///org/gnome/shell/ui/extensionSystem.js:59:14
_initializeUI@resource:///org/gnome/shell/ui/main.js:309:22
start@resource:///org/gnome/shell/ui/main.js:186:5
@resource:///org/gnome/shell/ui/init.js:6:17
```

**Comment by vanvugt** [comment]
The error was introduced in gjs 1.75.2: https://gitlab.gnome.org/GNOME/gjs/-/commit/a77a3b990 and is being tracked in https://gitlab.gnome.org/GNOME/gjs/-/issues/519

Seems to be a fundamental incompatibility between the C and JavaScript languages. Our only real option is to revert the above change in gjs or soften it to a warning, accepting/mitigating potential memory leaks. Other classes could in theory suffer logic errors from this if the result of a function changes over time, but that doesn't appear likely with `GAppInfo`. So we're only talking about preventing or limiting memory leaks here.

There's nothing the extension can do to avoid this short of deleting its usage of `Gio.AppInfo`. So I suggest the only way forward is to fix this in GJS (to mitigate memory leaks and not treat it as an error - again).

**Comment by vanvugt** [comment]
@jbicha, if you want something to release in the meantime then just patching a revert of https://gitlab.gnome.org/GNOME/gjs/-/commit/a77a3b990 should be OK. At worst that should take gjs back to the state in which we use it today.

**Comment by jbicha** [comment]
@vanvugt I believe Marco fixed gjs for Debian Experimental and Ubuntu 23.04.

**Comment by yochananmarqos** [comment]
FYI,  It's fixed with the latest commits to [MR 823](https://gitlab.gnome.org/GNOME/gjs/-/merge_requests/823).
